x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/import_rules_route.test.ts

```diff
@@ -0,0 +1,399 @@
+    });
+  });
+
+  describe('validation', () => {
+    test('returns reported error if index does not exist', async () => {
+      clients.clusterClient.callAsCurrentUser.mockResolvedValue(getEmptyIndex());
+      clients.alertsClient.find.mockResolvedValue(getFindResult());
+      clients.alertsClient.get.mockResolvedValue(getResult());
+      clients.alertsClient.create.mockResolvedValue(getResult());
+      const requestPayload = getSimpleRuleAsMultipartContent(['rule-1']);
+      const { statusCode, payload } = await server.inject(getImportRulesRequest(requestPayload));
+      const parsed: ImportSuccessError = JSON.parse(payload);
+
+      expect(parsed).toEqual({
+        errors: [
+          {
+            error: {
+              message:
+                'To create a rule, the index must exist first. Index undefined-undefined does not exist',
+              status_code: 409,
+            },
+            rule_id: 'rule-1',
+          },
+        ],
+        success: false,
+        success_count: 0,
+      });
+      expect(statusCode).toEqual(200);
+    });
+
+    test('returns 400 if file extension type is not .ndjson', async () => {
+      clients.alertsClient.find.mockResolvedValue(getFindResult());
+      clients.alertsClient.get.mockResolvedValue(getResult());
+      clients.alertsClient.create.mockResolvedValue(getResult());
+      const requestPayload = getSimpleRuleAsMultipartContent(['rule-1'], false);
+      const { statusCode, payload } = await server.inject(getImportRulesRequest(requestPayload));
+      const parsed: ImportSuccessError = JSON.parse(payload);
+
+      expect(parsed).toEqual({
+        message: 'Invalid file extension .json',
+        status_code: 400,
+      });
+      expect(statusCode).toEqual(400);
+    });
+  });
+
+  describe('single rule import', () => {
+    test('returns 200 if rule imported successfully', async () => {
+      clients.alertsClient.find.mockResolvedValue(getFindResult());
+      clients.alertsClient.get.mockResolvedValue(getResult());
+      clients.alertsClient.create.mockResolvedValue(getResult());
+      const requestPayload = getSimpleRuleAsMultipartContent(['rule-1']);
+      const { statusCode, payload } = await server.inject(getImportRulesRequest(requestPayload));
+      const parsed: ImportSuccessError = JSON.parse(payload);
+
+      expect(parsed).toEqual({
+        errors: [],
+        success: true,
+        success_count: 1,
+      });
+      expect(statusCode).toEqual(200);
+    });
+
+    test('returns reported conflict if error parsing rule', async () => {
+      const multipartPayload =
+        '--frank_is_awesome\r\n' +
+        `Content-Disposition: form-data; name="file"; filename="rules.ndjson"\r\n` +
+        'Content-Type: application/octet-stream\r\n' +
+        '\r\n' +
+        '{"name"::"Simple Rule Query","description":"Simple Rule Query","risk_score":1,"rule_id":"rule-1","severity":"high","type":"query","query":"user.name: root or user.name: admin"}\r\n' +
```

Comment:
Looks great!

Only optional I can think of is if were to pull out this one string and make it a variable like so?

```ts
      // This is not parseable due to the extra colon in it: "name"::"
      const unparseableLine =
        '{"name"::"Simple Rule Query","description":"Simple Rule Query","risk_score":1,"rule_id":"rule-1","severity":"high","type":"query","query":"user.name: root or user.name: admin"}\r\n';
      const multipartPayload = `${'--frank_is_awesome\r\n' +
        `Content-Disposition: form-data; name="file"; filename="rules.ndjson"\r\n` +
        'Content-Type: application/octet-stream\r\n' +
        '\r\n'}${unparseableLine}--frank_is_awesome--\r\n`;
```

The reason I just point that out, is to make the `::` stand out as an intentful parse error test for someone else later looking at the unit test. It can be easy to miss where the parse error is for future maintainers.

Reply:
I pulled this out into the same new utils file under `__mocks__` I'd mentioned above since I use it in a few places. Let me know if you think that the test utils belong in a different location.